File: 2487-optimal-partition-of-string/2487-optimal-partition-of-string.java
class Solution {
    public int partitionString(String s) {
        int de = 1;
        String su = "";
        for(char i : s.toCharArray()){
            if(su.indexOf(i) != -1){
                de++;
                su = "";
            }
            su += i;
        }
        return de;
        
    }
}

File: 2487-optimal-partition-of-string/2487-optimal-partition-of-string.py
class Solution(object):
    def partitionString(self, s):
        """
        :type s: str
        :rtype: int
        """
        st=""
        de=1
        for i in s:
            if i in st:
                st=""
                de+=1
                st = st + i
            else: st = st + i     
        return de